### DISCUSSION TITLE
SIMD

### DISCUSSION THREAD

**Discussion by Boshen** [oxc-project/oxc#7] [Ideas]
Topic around using SIMD for performance.

**Comment by Boshen** [comment]
Whitespace skipping.

Backround reading:
* https://rapidjson.org/md_doc_internals.html#SkipwhitespaceWithSIMD 
* https://github.com/Tencent/rapidjson/blob/bdc49ad80a3cff89a25839dcdcd6a68b457ec9f1/include/rapidjson/reader.h#L258-L517
* https://lemire.me/blog/2017/01/20/how-quickly-can-you-remove-spaces-from-a-string

**Comment by Boshen** [comment]
The results are promising from 
* https://github.com/Boshen/oxc/pull/26
* https://github.com/Boshen/oxc/pull/23

**Comment by strager** [comment]
In addition to lexing, I have considered using SIMD for line number calculations. I prototyped it a while ago (September 2020) but I didn't land it for some reason. https://github.com/quick-lint/quick-lint-js/commit/c930a9fe8e80a5d7fee4cace744d7e11701a12b3

SIMD can also be used for keyword recognition. (All keywords are 16 bytes or less.) I am currently making a YouTube video which explains this.

**Reply by Boshen** [reply]
Interesting! I can't wait for the video! I tried to understand multi-byte checking in SIMD but I just couldn't get my head around it.
